### DISCUSSION TITLE
Modifying User Table Fields

### DISCUSSION THREAD

**Discussion by hyphen81** [filamentphp/filament#8817] [Help]
I tried to modify fields on my user table as I'm trying to convert an existing app over to Filament. However, it doesn't seem like you can modify the user table fields as I'm now getting an error `Filament\FilamentManager::getUserName(): Return value must be of type string, null returned`.

Upon inspecting the FilemantManager file, I'm seeing this method: 

```
public function getUserName(Model | Authenticatable $user): string
    {
        if ($user instanceof HasName) {
            return $user->getFilamentName();
        }

        return $user->getAttributeValue('name');
    }
```

This seems like "name" is hard-coded and can't be modified without changing the source files, or overriding/extending the FilamentManager file. Is that what I'm supposed to do?  I can't imagine I'm the only person who's wanted to change the fields on the users table. Please help me understand how I can get the desired functionality without modifying the source files.

**Comment by hyphen81** [comment]
Just found this in another question:

https://filamentphp.com/docs/2.x/admin/users#configuring-the-name-attribute

Don’t see this referenced in the v3 docs.

**Reply by leandrocfe** [reply]
[here](https://filamentphp.com/docs/3.x/panels/users#configuring-the-users-name-attribute)

**Comment by youcefkacemDev** [comment]
I get the same error I changed the users table to be firstname and last name 
now I am going to change them back to name and add  username in my application

**Comment by encoderuz** [comment]
I get the same error and I implemented `HasName` interface like this:
`class User extends Authenticatable implements FilamentUser, HasName{
...
}`
